bundles/org.openhab.binding.mpd/src/main/java/org/openhab/binding/mpd/internal/discovery/MPDDiscoveryParticipant.java

```diff
@@ -0,0 +1,111 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.mpd.internal.discovery;
+
+import java.net.Inet4Address;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+import javax.jmdns.ServiceInfo;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.config.discovery.DiscoveryResult;
+import org.eclipse.smarthome.config.discovery.DiscoveryResultBuilder;
+import org.eclipse.smarthome.config.discovery.mdns.MDNSDiscoveryParticipant;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.openhab.binding.mpd.internal.MPDBindingConstants;
+import org.osgi.service.component.annotations.Component;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Implementation of {@link MDNSDiscoveryParticipant} that will discover Music Player Daemons.
+ *
+ * @author Stefan Röllin - Initial contribution
+ *
+ */
+@NonNullByDefault
+@Component(immediate = true)
+public class MPDDiscoveryParticipant implements MDNSDiscoveryParticipant {
+
+    private final Logger logger = LoggerFactory.getLogger(MPDDiscoveryParticipant.class);
+
+    @Override
+    public Set<ThingTypeUID> getSupportedThingTypeUIDs() {
+        return Collections.singleton(MPDBindingConstants.THING_TYPE_MPD);
+    }
+
+    @Override
+    public String getServiceType() {
+        return "_mpd._tcp.local.";
+    }
+
+    @Override
+    @Nullable
+    public DiscoveryResult createResult(ServiceInfo service) {
+        ThingUID uid = getThingUID(service);
+        String host = getHostAddress(service);
+        int port = service.getPort();
+
+        logger.debug("Music Player Daemon found on host {} port {}", host, port);
+
+        if (uid == null || host == null || "".equals(host)) {
+            return null;
+        }
+
+        final Map<String, Object> properties = new HashMap<>(2);
+        properties.put(MPDBindingConstants.PARAMETER_IPADDRESS, host);
+        properties.put(MPDBindingConstants.PARAMETER_PORT, port);
+
+        String name = service.getName();
+
+        final DiscoveryResult result = DiscoveryResultBuilder.create(uid).withProperties(properties).withLabel(name)
+                .build();
```

Comment:
Can you add a representation property?

Reply:
Did not know of the representation property. I have added it.